### ISSUE TITLE
Looking for package hierarchy guidance. Where are the conversion utils?

### ISSUE THREAD

**Issue by etjones** [robbykraft/Origami#60]
**TL;DR: where is `SVGtoFOLD()` in the Node package?  (Also, what's the best way to go about finding things for myself?)**

Man, Rabbit Ear/ Origami does a lot of stuff! I've been watching the repo this summer and I really admired how much work you did.

I'm having a hard time figuring it all out, though, and finding utilities I know are there. 

Today's problem: 
I'm looking to convert some SVG files to FOLD files suitable for use with Amanda Ghassei's [Origami Simulator](http://apps.amandaghassaei.com/OrigamiSimulator/).  Using the source repo, I was able to call `SVGtoFOLD()` in a Node REPL. Super useful!  But, when did `npm install rabbit-ear` and then tried to find that function in the hierarchy in the Rollup-built package, I struck out.

So the immediate question is: what's a valid Node import that would let me do something like: 
``` node
const rabbit_ear = require('rabbit-ear');
svg_path = '<PATH/TO/file.svg>'
fold_path = '<PATH/TO/file.fold>'
// Where does SVGtoFOLD() live in the hierarchy?
rabbit_ear.<WHICH_PACKAGE>.<WHICH_OTHER_PACKAGE>.SVGtoFOLD(svg_path, fold_path);
```
The  longer-term question is probably about discoverability. Are there any docs that would allow users to find where all the (many, and awesome) capabilities of RE/Origami live?

Thanks!

**Comment by phauly** [comment]
I had the same question! I would love to start playing with rabbitear but I'm having troubles in figuring out which methods are where.

**Comment by etjones** [comment]
I did find an answer to my particular question in one of the test files, so [that](https://github.com/robbykraft/Origami/tree/master/tests) seems like a good place to start. I suspect that for something more comprehensive, one of us users might have to start volunteering to write some docs. One of these days, I'm sure I'll hop in there. One of these days... ;-)

**Comment by phauly** [comment]
me too, one of these days ... ;)

**Comment by phauly** [comment]
Wow! Already writing a page (with code) in which there are few example on how to create crease patters with RabbitEar, export them so that they are importable without error into Amanda's Origami Simulator ( https://origamisimulator.org ) would be extremely useful! I tried to do it but failed ;(

**Comment by freestraws** [comment]
Is there any specific error you get when it fails?

**Comment by phauly** [comment]
When I wrote "failed", I referred to the fact I was not able to run code for saving a simple .svg file. Sorry if this was not clear.

**Comment by freestraws** [comment]
ah, ok, I've done it before for a few files that I then used to make obj files with the simulator, I can try to find the setup I had for it when I get home.
I think I did it with an interactive crease pattern test html that was in rabbit-ear a while ago, but that test page was taken out in later versions for some reason (might've been too buggy with some edge cases, but it worked for my purposes and was good to show me a bunch of different rabbit-ear functions' use cases)

**Comment by phauly** [comment]
If you have time to put online how to create your setup, it would be awesome!

**Comment by freestraws** [comment]
I couldn't figure it out again (the page i thought it would be on just wouldn't load) so I tried to use the examples/editor/editor.html in the most recent version, was having difficulty with it... It looks like that would work if I could get it running, but I haven't found any instructions for running the examples other than just running rollup to set it up...

**Comment by yllennods** [comment]
ok, for node you can do something like:
```
let origami = rabbit.Origami();
fs.readFile("crane.svg", 'utf8', function(err, data){
    origami.load(data);
});
let fold = origami.export.fold();
```
and as long as "crane.svg" exists in the same directory as the script you're running, `fold` will contain a string that is the fold equivalent of the svg.  You could save that to a file and load it up in Origami Simulator.  I've noticed that some crease patterns don't work out of the box, like crane for example has some issues upon import because there are faces with 5+ edges which Origami Simulator doesn't know how to deal with.  There is no direct solution to that for fold files at least, but for svg, in "File Import Tips" you can see that that's what the yellow lines are for, splitting those faces up into <5 edged faces.
You can also do the reverse (give a fold file, get an svg with origami.export.svg()) but you may have to mess with the svg to get something the Origami Simulator will recognize, like if the colors are different from what's in her "File Import tips", you can manually go in and fix those colors.  Or string replace them with the correct colors before saving the file, whatever.

**Comment by freestraws** [comment]
I figured it out.
For node you can do something like:
```
let origami = rabbit.Origami();
fs.readFile("crane.svg", 'utf8', function(err, data){
    origami.load(data);
});
let fold = origami.export.fold();
```
and as long as "crane.svg" exists in the same directory as the script you're running, `fold` will be a string that is the fold equivalent of the svg.  You could save that to a file and load it up in Origami Simulator.  I've noticed that some crease patterns don't work out of the box, for example [this crane](https://github.com/robbykraft/Origami/blob/master/files/svg/crane.svg) has some issues upon import because there are faces with 5+ edges which Origami Simulator doesn't  directly know how to deal with.  There is no direct solution to that for fold files at least, but for svg, in "File Import Tips" you can see that that's what the yellow lines are for, splitting those faces up into <5 edged faces.

You can also do the reverse (give a fold file, get an svg with origami.export.svg()) but you may have to mess with the svg to get something the Origami Simulator will recognize, like if the colors are different from what's in her "File Import tips", you can manually go in and fix those colors.  Or string replace them with the correct colors before saving the file, whatever.

(I accidentally commented from my work account at first, aaah)

**Comment by robbykraft** [comment]
Hello all, please let me offer my deepest apologies for not being responsive. Things have not been great for more than a year... I'm very sorry.

In good news I remained hard at work, and the entire project has been re-written from the ground up. I took your suggestion @etjones and made the readme speak to developers not users, there is a summary of the src/ folder, and i hope to expand this with more detail. https://github.com/robbykraft/Origami/blob/e83d5a7dc22d2ba5d459cc33a04cec453138ffca/readme.md

i know it's too late but let me address each of your questions anyway.

at least now, [fold-to-svg](https://github.com/robbykraft/fold-to-svg) should by default output red and blue lines for mountain and valley, which is what origami simulator is looking for. But there is a more direct pathway - by sending it a FOLD object. The newly-updated docs demonstrate this under the Crease Pattern section https://rabbitear.org/docs/origami.html#crease-pattern The instance of Origami Simulator I'm using is a re-write I did and it's up here https://github.com/robbykraft/OrigamiSimulator, the re-write just unhooked all front-end from the engine so it can run even in Node. i'm sorry to say i haven't updated it in a while and as i was implementing this for the docs i realized it could use some updating. @phauly this workflow might be what you are talking about. I would love to:

1. clean up my Origami Simulator instance.
2. expand this little example on the docs to be a full blown crease pattern - origami simulator editor. From the last version of Rabbit Ear before the re-write I had an example of this: cp.rabbitear.org, bit this needs updating now that the core has been re-written too.

these kinds of conversations really help me get organized. I'm sorry if my absence was discouraging. The project is back online and me with it. I would be so excited for more conversations like these in the future.
